packages/types/src/auth.ts

```diff
@@ -0,0 +1,39 @@
+import { Provider } from "./util";
+
+export interface Identity {
+  /**
+   * A {Date} when the identity or credential will no longer be accepted.
+   */
+  readonly expiration?: Date;
+}
+
+/**
+ * An object representing temporary or permanent AWS credentials.
+ */
+export interface Credentials extends Identity {
```

Comment:
Give it a new name, like `AwsCredentialIdentity`
```suggestion
interface AwsCredentialIdentity extends Identity {
```

We should continue to export `Credentials` from `credentials.ts` for backward compatibility.